implementation_design.md

```diff
@@ -219,35 +74,81 @@ Error types:
 - WRONG_CHAIN_ID: wrong chain id for the target chain
 - WRONG_SIGNATURE: signature failed to verify
 - BAD_ROUND_NUMBER: incorrect round, such as one from the past
+- RESERVE_PRICE_NOT_MET: bid amount does not meet the minimum required reserve price onchain
 - INSUFFICIENT_BALANCE: the bid amount specified in the request is higher than the deposit balance of the depositor in the contract
 
-## Auctioneer Implementation
+## Sequencer API
 
-Rounds, to prevent the need for synchronization, are set to be at the each minute boundary based on unix time. 
+The sequencer must expose a new RPC namespace called `timeboost` and a RPC method called `timeboost_sendExpressLaneTransaction`
 
-#### Auction State
+### `timeboost_sendExpressLaneTransaction`
 
-Add an `expressLaneAuctionState` struct to the sequencer with the following:
-
-```go
-type expressLaneAuctionState {
+```json
+{
+  "type": "object",
+  "properties": {
+    "chainId": {
+      "type": "bigInt",
+      "description": "chain id of the target chain"
+    },
+    "round": {
+      "type": "uint64",
+      "description": "round number (0-indexed) for the round the bidder wants to become the controller of"
+    },
+    "auctionContractAddress": {
+      "type": "address",
+      "description": "hex string of the auction contract address that the bid corresponds to"
+    },
+    "sequenceNumber": {
+      "type": "uint64",
+      "description": "the per-round nonce of express lane submissions. Each submission to the express lane during a round increases this sequence number by one, and if submissions are received out of order, the sequencer will queue them for processing in order. This is reset to 0 at each round"
+    },
+    "transaction": {
+      "type": "bytes",
+      "description": "hex string of the RLP encoded transaction payload that submitter wishes to be sequenced through the express lane"
+    },
+    "options": {
+      "type": "ArbitrumConditionalOptions",
+      "description": "conditional options for Arbitrum transactions, supported by normal sequencer endpoint https://github.com/OffchainLabs/go-ethereum/blob/48de2030c7a6fa8689bc0a0212ebca2a0c73e3ad/arbitrum_types/txoptions.go#L71"
+    },
+    "signature": {
+      "type": "bytes",
+      "description": "Ethereum signature over the bytes encoding of (keccak256(TIMEBOOST_BID), padTo32Bytes(chainId), auctionContractAddress, uint64ToBytes(round), uint64ToBytes(sequenceNumber), transaction)"
+    }
+  },
 }
 ```
 
-#### Processing Bids 
+The sequencer will perform the following validation:
+- check the fields are not nil
+- check if the chain id is correct
+- check if the auction contract address is correct
+- check the current round has an express lane controller
+- check if the bid is intended for the current round
+- verify the signature and recover the sender address
```

Comment:
Probably better not to call this the "sender" because it is not the sender of the transaction. (The transaction might be signed by a different party.). I would say "recover the signer address" here; then in the next item "verify the signer is ..."